examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts

```diff
@@ -42,47 +57,153 @@ export const observationCategoryMapping: Record<string, CodeableConcept> = {
   },
 };
 
+export const consentScopeMapping: Record<string, CodeableConcept> = {
+  adr: {
+    coding: [
+      {
+        system: HTTP_TERMINOLOGY_HL7_ORG + '/CodeSystem/consentscope',
+        code: 'adr',
+        display: 'Advanced Care Directive',
+      },
+    ],
+  },
+  patientPrivacy: {
+    coding: [
+      {
+        system: HTTP_TERMINOLOGY_HL7_ORG + '/CodeSystem/consentscope',
+        code: 'patient-privacy',
+        display: 'Patient Privacy',
+      },
+    ],
+  },
+  treatment: {
+    coding: [
+      {
+        system: HTTP_TERMINOLOGY_HL7_ORG + '/CodeSystem/consentscope',
+        code: 'treatment',
+        display: 'Treatment',
+      },
+    ],
+  },
+};
+
+export const consentCategoryMapping: Record<string, CodeableConcept> = {
+  acd: {
+    coding: [
+      {
+        system: HTTP_TERMINOLOGY_HL7_ORG + '/CodeSystem/consentcategorycodes',
+        code: 'acd',
+        display: 'Advanced Care Directive',
+      },
+    ],
+  },
+  nopp: {
+    coding: [
+      {
+        system: HTTP_TERMINOLOGY_HL7_ORG + '/CodeSystem/v3-ActCode',
+        code: 'nopp',
+        display: 'Notice of Privacy Practices',
+      },
+    ],
+  },
+  pay: {
+    coding: [
+      {
+        system: HTTP_TERMINOLOGY_HL7_ORG + '/CodeSystem/v3-ActCode',
+        code: 'pay',
+        display: 'Payment',
+      },
+    ],
+  },
+  med: {
+    coding: [
+      {
+        system: HTTP_TERMINOLOGY_HL7_ORG + '/CodeSystem/v3-ActCode',
+        code: 'med',
+        display: 'Medical',
+      },
+    ],
+  },
+};
+
+export const consentPolicyRuleMapping: Record<string, CodeableConcept> = {
+  hipaaNpp: {
+    coding: [
+      {
+        system: HTTP_TERMINOLOGY_HL7_ORG + '/CodeSystem/consentpolicycodes',
+        code: 'hipaa-npp',
+        display: 'HIPAA Notice of Privacy Practices',
+      },
+    ],
+  },
+  hipaaSelfPay: {
+    coding: [
+      {
+        system: HTTP_TERMINOLOGY_HL7_ORG + '/CodeSystem/consentpolicycodes',
+        code: 'hipaa-self-pay',
+        display: 'HIPAA Self-Pay Restriction',
+      },
+    ],
+  },
+};
+
+/**
+ * This function takes data about an Observation and creates or updates an existing
+ * resource with the same patient and code.
+ *
+ * @param medplum - A Medplum client
+ * @param patient - A Patient resource that will be stored as the subject
+ * @param code - A code for the observation
+ * @param category - A category for the observation
+ * @param valueXAttribute - The value[x] field where the answer should be stored
+ * @param value - The value to be stored in the observation
+ */
 export async function upsertObservation(
   medplum: MedplumClient,
   patient: Patient,
   code: CodeableConcept,
   category: CodeableConcept,
-  valueCoding: QuestionnaireResponseItemAnswer | undefined
+  valueXAttribute: 'valueCodeableConcept' | 'valueDateTime',
```

Comment:
This parameter needs a better name and a better type

```suggestion
  answerType: QuestionnaireItemType,
```